table/server/master/src/main/java/alluxio/master/table/DbConfig.java

```diff
@@ -55,153 +97,535 @@ public DbConfig(@JsonProperty("bypass") @Nullable BypassEntry bypassEntry) {
+        throws IOException, JsonProcessingException {
+      TablesEntryDeserializer<NamePatternEntry> deserializer = new TablesEntryDeserializer<>();
+      IncludeExcludeList<NamePatternEntry> list =
+          deserializer.deserializeToList(NamePatternEntry.class, jp, cxt);
+      return new IgnoreTablesSpec(list);
+    }
+  }
+
+  static class BypassTablesSpecDeserializer extends JsonDeserializer<BypassTablesSpec> {
+    @Override
+    public BypassTablesSpec deserialize(JsonParser jp, DeserializationContext cxt)
+        throws IOException, JsonProcessingException {
+      TablesEntryDeserializer<TableEntry> deserializer = new TablesEntryDeserializer<>();
+      IncludeExcludeList<TableEntry> list =
+          deserializer.deserializeToList(TableEntry.class, jp, cxt);
+      return new BypassTablesSpec(list);
+    }
+  }
+
+  /**
+   * On top of a regular NamePatternEntry, contains additional partition specification.
+   */
+  @JsonDeserialize(using = TableEntryDeserializer.class)
+  public static class TableEntry extends NamePatternEntry {
+    private final IncludeExcludeList<NamePatternEntry> mPartitions;
 
     /**
+     * Creates an instance with a specific table name and no partition specification.
+     *
      * @param tableName table name
-     * @param partitions partition names
      */
-    @JsonCreator
-    public BypassTableEntry(@JsonProperty("table") String tableName,
-                            @JsonProperty("partitions") Set<String> partitions) {
-      Preconditions.checkArgument(!tableName.isEmpty(), "empty table name");
-      mTableName = tableName;
+    public TableEntry(String tableName) {
+      this(tableName, IncludeExcludeList.empty());
+    }
+
+    /**
+     * Creates an instance with a specific table name and possibly partitions specifications.
+     *
+     * @param tableName table name
+     * @param partitions partitions
+     */
+    public TableEntry(String tableName, IncludeExcludeList<NamePatternEntry> partitions) {
+      super(tableName);
       mPartitions = partitions;
     }
 
     /**
-     * @return table name
+     * Creates an instance from a {@link NamePatternEntry} with no partition specification.
+     * @param namePatternEntry name entry
      */
-    public String getTable() {
-      return mTableName;
+    public TableEntry(NamePatternEntry namePatternEntry) {
+      super(namePatternEntry);
+      mPartitions = IncludeExcludeList.empty();
     }
 
     /**
-     * @return partition names
+     * Returns partition specifications.
+     * @return partitions
      */
-    public Set<String> getPartitions() {
+    public IncludeExcludeList<NamePatternEntry> getPartitions() {
       return mPartitions;
     }
 
+    /**
+     * Returns table name if the entry is not a regex entry.
+     * @return table name, null if the entry is a regex entry
+     */
+    @Nullable
+    public String getTable() {
+      return getName();
+    }
+
     @Override
     public boolean equals(Object other) {
```

Comment:
Trying to remove this override gives me this error:

```
[INFO] --- spotbugs-maven-plugin:4.0.4:check (default) @ alluxio-table-server-master ---
[INFO] BugInstance size is 1
[INFO] Error size is 0
[INFO] Total bugs: 1
[ERROR] Medium: alluxio.master.table.DbConfig$TableEntry doesn't override DbConfig$NamePatternEntry.equals(Object) [alluxio.master.table.DbConfig$TableEntry] At DbConfig.java:[line 1] EQ_DOESNT_OVERRIDE_EQUALS
```

Maybe we should leave it as

```java
@Override
public boolean equals(Object other) {
  return super.equals(other);
}
```

so that we make our intention clear that we do not want to compare the partitions.

FYI @maobaolong .